compiler/control/CompileMethod.cpp

```diff
@@ -200,6 +200,22 @@ registerTrampoline(uint8_t *start, uint32_t size, const char *name)
       writePerfToolEntry(start, size, name);
    }
 
+static void
+runDiagnostics(TR::Compilation * comp, const char * reason)
```

Comment:
Given that diagnostic already means something specific in TR, should we consider a different name for this function?